aws/kinesis.cpp

```diff
@@ -233,6 +234,9 @@ get_shard_iterator(simple_consumer *sc, const Aws::String &shardId,
   auto outcome = sc->client->GetShardIterator(sir);
   if (!outcome.IsSuccess()) {
     auto e = outcome.GetError();
+    if (!e.ShouldRetry()) {
+      sequenceId.clear(); // issue 354 invalid checkpoint clear/log and retry
```

Comment:
Is there any cases where we wouldn't want to clear on a non-retryable error in `GetShardIterator`?

Reply:
None that I know of so the only thing we can do here is drop that state, log the error and try again.